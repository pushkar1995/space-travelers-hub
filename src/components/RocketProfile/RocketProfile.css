.rocket-profile-table {
  width: 40vw;
  border-collapse: collapse;
  border: 1px solid black;
  margin: 20px;
  padding: 10px;
}

h2 {
  margin: 20px;
}

.rocket-profile-table th,
.rocket-profile-table td {
  border: 1px solid black;
  padding: 10px;
  text-align: left;
  font-size: 1.2rem;
  font-weight: 500;
}

.no-rocket-placeholder {
  font-weight: 600;
  background-color: #f0f0f0;
  border-radius: 4px;
  padding: 10px;
}
