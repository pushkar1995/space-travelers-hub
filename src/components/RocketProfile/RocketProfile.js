import React from 'react';
import { useSelector } from 'react-redux';
import './RocketProfile.css';

function RocketProfile() {
  const rockets = useSelector((state) => state.rockets.rockets);
  const bookedRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <div>
      <h2>My Rockets</h2>
      <table className="rocket-profile-table">
        <tbody>
          {bookedRockets.length === 0 ? (
            <tr>
              <td className="no-rocket-placeholder">No Rockets Added</td>
            </tr>
          ) : (
            bookedRockets.map((rocket) => (
              <tr key={rocket.id}>
                <td>{rocket.name}</td>
              </tr>
            ))
          )}
        </tbody>
      </table>
    </div>
  );
}

export default RocketProfile;
