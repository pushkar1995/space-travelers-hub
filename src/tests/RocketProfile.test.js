import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import RocketProfile from '../components/RocketProfile/RocketProfile';

const mockStore = configureStore([]);

describe('RocketProfile', () => {
  it('should display "No Rockets Added" when there are no rockets booked', () => {
    const initialState = {
      rockets: {
        rockets: [],
      },
    };

    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <RocketProfile />
      </Provider>,
    );

    const noRockets = getByText('No Rockets Added');
    expect(noRockets).toMatchSnapshot();
  });

  it('should display all rockets that have been booked', () => {
    const initialState = {
      rockets: {
        rockets: [
          { id: 1, name: 'Falcon 1', reserved: true },
          { id: 2, name: 'Falcon 9', reserved: true },
        ],
      },
    };

    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <RocketProfile />
      </Provider>,
    );

    const rocket1Text = getByText('Falcon 1');
    const rocket2Text = getByText('Falcon 9');
    expect(rocket1Text).toMatchSnapshot();
    expect(rocket2Text).toMatchSnapshot();
  });
});
