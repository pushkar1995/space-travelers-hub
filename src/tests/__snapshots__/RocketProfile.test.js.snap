// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketProfile should display "No Rockets Added" when there are no rockets booked 1`] = `
<td
  class="no-rocket-placeholder"
>
  No Rockets Added
</td>
`;

exports[`RocketProfile should display all rockets that have been booked 1`] = `
<td>
  Falcon 1
</td>
`;

exports[`RocketProfile should display all rockets that have been booked 2`] = `
<td>
  Falcon 9
</td>
`;
